pkg/devfile/parser/parse.go

```diff
@@ -212,11 +250,36 @@ func parseFromURI(uri string, curDevfileCtx devfileCtx.DevfileCtx) (DevfileObj,
 		d.Ctx = devfileCtx.NewURLDevfileCtx(u.String())
 	}
 	d.Ctx.SetURIMap(curDevfileCtx.GetURIMap())
-	// Fill the fields of DevfileCtx struct
-	err = d.Ctx.PopulateFromURL()
-	if err != nil {
-		return DevfileObj{}, err
+	return populateAndParseDevfile(d, true)
+}
+
+func parseFromRegistry(registryId, registryURL string, curDevfileCtx devfileCtx.DevfileCtx) (DevfileObj, error) {
+	if registryURL != "" {
+		devfileContent, err := getDevfileFromRegistry(registryId, registryURL)
+		if err != nil {
+			return DevfileObj{}, err
+		}
+		return ParseDevfile(ParserArgs{data: devfileContent, registryURLs: curDevfileCtx.GetRegistryURLs()})
+	} else if curDevfileCtx.GetRegistryURLs() != nil {
+		for _, registry := range curDevfileCtx.GetRegistryURLs() {
+			devfileContent, err := getDevfileFromRegistry(registryId, registry)
+			if devfileContent != nil && err == nil {
+				return ParseDevfile(ParserArgs{data: devfileContent, registryURLs: curDevfileCtx.GetRegistryURLs()})
+			}
```

Comment:
no we shouldn't. as the registryURLs in ctx is an array, the `parentId` will be valid for only one registry
we should parse from that registry if it's valid, or we should continue try the next. 
if none of the registryURL is valid, it will return error at the end of the function